### ISSUE TITLE
Failing tests

### ISSUE THREAD

**Issue by FRidh** [countergram/pytidylib#13]
A bunch of tests seem to fail.
```
..FF.F.FF...F..
======================================================================
FAIL: test_alt_added_to_img (tests.test_fragments.TestFrags1)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python2.7-pytidylib-0.2.4.drv-0/pytidylib-0.2.4/tests/test_fragments.py", line 41, in test_alt_added_to_img
    self.assertEqual(doc, expected)
AssertionError: u"<img src='foo'>" != u'<img src=\'foo\' alt="">'
- <img src='foo'>
+ <img src='foo' alt="">
?               +++++++


======================================================================
FAIL: test_entity_preserved_using_bytes (tests.test_fragments.TestFrags1)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python2.7-pytidylib-0.2.4.drv-0/pytidylib-0.2.4/tests/test_fragments.py", line 47, in test_entity_preserved_using_bytes
    self.assertEqual(doc, expected)
AssertionError: '\xc3\xa9' != '&eacute;'

======================================================================
FAIL: test_numeric_entities_using_bytes (tests.test_fragments.TestFrags1)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python2.7-pytidylib-0.2.4.drv-0/pytidylib-0.2.4/tests/test_fragments.py", line 53, in test_numeric_entities_using_bytes
    self.assertEqual(doc, expected)
AssertionError: '\xc3\xa9' != '&#233;'

======================================================================
FAIL: test_alt_added_to_img (tests.test_docs.TestDocs1)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python2.7-pytidylib-0.2.4.drv-0/pytidylib-0.2.4/tests/test_docs.py", line 53, in test_alt_added_to_img
    self.assertEqual(doc, expected)
AssertionError: u'<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">\n<html>\n  <head>\n    <tit [truncated]... != u'<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">\n<html>\n  <head>\n    <tit [truncated]...
  <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">
  <html>
    <head>
      <title></title>
    </head>
    <body>
-     <img src='foo'>
+     <img src='foo' alt="">
?                   +++++++
    </body>
  </html>


======================================================================
FAIL: test_entity_preserved_using_bytes (tests.test_docs.TestDocs1)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python2.7-pytidylib-0.2.4.drv-0/pytidylib-0.2.4/tests/test_docs.py", line 59, in test_entity_preserved_using_bytes
    self.assertEqual(doc, expected)
AssertionError: '<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">\n<html>\n  <head>\n    <title></title>\n  </head>\n  <body>\n    \xc3\xa9\n  </body>\n</html>\n' != '<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">\n<html>\n  <head>\n    <title></title>\n  </head>\n  <body>\n    &eacute;\n  </body>\n</html>\n'

======================================================================
FAIL: test_numeric_entities_using_bytes (tests.test_docs.TestDocs1)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python2.7-pytidylib-0.2.4.drv-0/pytidylib-0.2.4/tests/test_docs.py", line 65, in test_numeric_entities_using_bytes
    self.assertEqual(doc, expected)
AssertionError: '<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">\n<html>\n  <head>\n    <title></title>\n  </head>\n  <body>\n    \xc3\xa9\n  </body>\n</html>\n' != '<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">\n<html>\n  <head>\n    <title></title>\n  </head>\n  <body>\n    &#233;\n  </body>\n</html>\n'

----------------------------------------------------------------------
Ran 15 tests in 0.078s

FAILED (failures=6)
```

**Comment by countergram** [comment]
What version of tidy library is this with?

**Comment by FRidh** [comment]
pytidylib-0.2.4

**Comment by countergram** [comment]
Sorry, I mean the underlying tidy library which will be installed as a system package somewhere.

**Comment by FRidh** [comment]
It seems to be 5.0.0.

**Comment by countergram** [comment]
Latest git master should be working with 5 but I need to check again. Due to a new job I didn’t get it fully packaged yet.

**Comment by jbicha** [comment]
I get the same test failures with pytidylib 0.2.4 and tidy-html 5.2.0 even after applying your 2 2015 commits as patches.

**Comment by mitya57** [comment]
I have done some debugging on this (tested with the latest git snapshot of tidy-html5):

* `test_entity_preserved_using_bytes` can be fixed by adding `'preserve-entities': 1` to the options (tidylib no longer preserves them by default);
* `test_numeric_entities_using_bytes` can be fixed by adding `'output-encoding': 'ascii'` to the options (tidylib no longer outputs numeric entities for unicode encodings);
* `test_alt_added_to_img` can not be easily fixed: since htacg/tidy-html5@18880eab5551dff6 tidylib ignores empty strings as options. Reverting that commit in tidylib makes the test pass again.
